pemFioi/conceptViewer-1.0-mobileFirst.js

```diff
@@ -439,29 +439,25 @@ function getConceptsFromBlocks(includeBlocks, allConcepts, context) {
 
   if(includeBlocks.generatedBlocks) {
     for(var genName in includeBlocks.generatedBlocks) {
-      var categoriesByBlocks = {};
+      // this variable is used in order to make sure that we don't include two
+      // times a documentation
       var includedCategories = [];
-      if(context && context.customBlocks && context.customBlocks[genName]) {
-        for(var catName in context.customBlocks[genName]) {
-          var categoryConceptName = genName + '_' + catName;
-          if(!allConceptsById[categoryConceptName]) { continue; }
-          var blockList = context.customBlocks[genName][catName];
-          for(var i=0; i<blockList.length; i++) {
-            categoriesByBlocks[blockList[i].name] = categoryConceptName;
+      var cleanedConcepts = cleanAllConcept(allConcepts);
+      for (var functionKey in includeBlocks.generatedBlocks[genName]) {
+        var functionName = includeBlocks.generatedBlocks[genName][functionKey];
+        var concept = findConceptByFunction(cleanedConcepts, functionName);
+        if (concept) {
+          // if we does not have the concept already pushed, we push it.
+          if (includedCategories.indexOf(concept.id) == -1) {
```

Comment:
includedCategories should be renamed as it's not categories anymore (includedConceptIds for instance).

Reply:
Yes, "includedConceptIds" was the default name but "includedConceptIds" make more sense